plugins/GSdx/Renderers/DX11/GSOsdManagerDX11.cpp

```diff
@@ -0,0 +1,266 @@
+#include "stdafx.h"
+#include "GSOsdManagerDX11.h"
+
+GSOsdManagerDX11::GSOsdManagerDX11(ID3D11Device *dev)
+{
+	// NOTE: workaround for Fraps/Afterburner
+	if (!m_log_enabled && !m_monitor_enabled)
+		return;
+
+	HRESULT hr = E_FAIL;
+
+	D2D1_FACTORY_OPTIONS d2d_factory_options = {};
+
+	#ifdef DEBUG
+		d2d_factory_options.debugLevel = D2D1_DEBUG_LEVEL_INFORMATION;
+	#endif
+
+	CComPtr<IDXGIDevice> dxgid;
+	CComPtr<ID2D1Factory1> d2d_factory;
+
+	hr = D2D1CreateFactory(
+		D2D1_FACTORY_TYPE_SINGLE_THREADED,
+		__uuidof(ID2D1Factory1),
+		&d2d_factory_options,
+		(void **)&d2d_factory
+	);
+
+	if (FAILED(hr))
+	{
+		fprintf(stderr, "NOTE: DX11 OSD can only be enabled on Windows 7 with the Platform Update.\n");
+		fprintf(stderr, "Disabling OSD will remove this warning.\n");
+		fprintf(stderr, "https://support.microsoft.com/en-us/help/2670838/platform-update-for-windows-7-sp1-and-windows-server-2008-r2-sp1\n");
+		return;
+	}
+
+	hr = dev->QueryInterface(IID_PPV_ARGS(&dxgid));
+
+	hr = DWriteCreateFactory(
+		DWRITE_FACTORY_TYPE_SHARED,
+		__uuidof(IDWriteFactory),
+		reinterpret_cast<IUnknown **>(&m_dwrite_factory)
+	);
+
+	hr = d2d_factory->CreateDevice(dxgid, &m_dev_2d);
+
+	m_dev_2d->CreateDeviceContext(
+		D2D1_DEVICE_CONTEXT_OPTIONS_NONE,
+		&m_ctx_2d
+	);
+
+	int r = theApp.GetConfigI("osd_color_r");
+	int g = theApp.GetConfigI("osd_color_g");
+	int b = theApp.GetConfigI("osd_color_b");
+
+	uint32 color = b | (g << 8) | (r << 16);
+
+	m_ctx_2d->CreateSolidColorBrush(
+		D2D1::ColorF(color, m_opacity),
+		&m_osd_color
+	);
+
+	m_dwrite_factory->CreateTextFormat(
+		L"tahoma",
+		nullptr,
+		DWRITE_FONT_WEIGHT_LIGHT,
+		DWRITE_FONT_STYLE_NORMAL,
+		DWRITE_FONT_STRETCH_NORMAL,
+		(float)m_size,
+		L"en-US", // TODO
+		&m_osd_text_format
+	);
+}
+
+void GSOsdManagerDX11::Log(const char *utf8, uint32 color)
+{
+	std::wostringstream s;
+
+	s << utf8;
```

Comment:
I don't think this will convert UTF-8 characters properly?

Reply:
A quick search and it seems `mbstowcs` is what needs to be done here.

If it's sounds correct to you I can write an implementation that uses `mbstowcs` with a `wchar` buffer of 256 in similar fashion to OGL.

Reply:
Maybe use `convert_utf8_to_utf16` from common/include/PluginCompatibility.h (header file should already be included). I don't think mbstowcs will work for UTF-8 conversion on Windows .